Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -4013,58 +4015,49 @@ - (void)disconnect
 - (void)metadataLoaded
 {
     ensureOnMainThread([self, strongSelf = retainPtr(self)] {
-        if (!m_player)
-            return;
-
-        m_player->queueTaskOnEventLoop([player = m_player] {
-            if (player)
+        if (RefPtr player = m_player.get()) {
+            player->queueTaskOnEventLoop([player = WTFMove(player)] {
```

Comment:
We are extending the lifetime of player, is it needed?
Ditto below or elsewhere probably.

Reply:
That's the aim of the patch, and why the object is now refcounted.